contracts/swaps/connectors/SwapsImplSovrynSwap.sol

```diff
@@ -253,13 +246,38 @@ contract SwapsImplSovrynSwap is State, ISwapsImpl {
         address sourceTokenAddress,
         address destTokenAddress,
         uint256 sourceTokenAmount,
-        address sovrynSwapContractRegistryAddress
+        address sovrynSwapContractRegistryAddress,
+        IERC20[] memory defaultPath
     ) public view returns (uint256 expectedReturn) {
         ISovrynSwapNetwork sovrynSwapNetwork =
             getSovrynSwapNetworkContract(sovrynSwapContractRegistryAddress);
+
         IERC20[] memory path =
-            sovrynSwapNetwork.conversionPath(IERC20(sourceTokenAddress), IERC20(destTokenAddress));
+            defaultPath.length >= 3
+                ? defaultPath
+                : sovrynSwapNetwork.conversionPath(
+                    IERC20(sourceTokenAddress),
+                    IERC20(destTokenAddress)
+                );
+
```

Comment:
why aren't you using getConversionPath here as well?

Reply:
If we go forward with this, there should be at least a warning on the documentation. It might be cleaner to implement this in a way that none of the functions assumes to have access to the protocol storage.
@tjcloa what do you think?

Reply:
Correct, this is the only function gets called directly instead of delegate call (by the `_swapsExpectedReturn` function).
I agree with adding the better & clearer comment for this part.
wdyt @tjcloa  ?

Reply:
as a quick solution i'm fine with adding clearer comments
but i agree with ororo that this design is in general confusing and can cause problems in the future
much clearer is to use logic contracts without caring abt their deployment addresses at all presuming it is used as a library with proxy storage only and all the calls are done via the proxy
i would prefer to either (on per-function basis):
- extract "direct" public functions like this into a separate contract or move it into an existing contract 
- use via delegatecall
depending on priorities and efforts to impelement this (which will require adapting tests too) a separate refactoring task can be added to monday.com
@ororopickpocket, @cwsnt wdyt?